### ISSUE TITLE
Avoid throwing an exception on start up

### ISSUE THREAD

**Issue by yurydelendik** [wasdk/WebAssemblyStudio#68]
When debugging with pause on all exception is perform, the exception is thrown when the application is loading. It is about unexpected token/character in JSON:

<img width="1121" alt="screen shot 2018-02-12 at 9 02 42 am" src="https://user-images.githubusercontent.com/1523410/36103055-dcae6902-0fd3-11e8-896e-ae6adbc666df.png">

It is expected that application will not throw any exception (even caught one) during startup.

(There is another one that is coming from json3 module when `npm run dev-server` is used. Let us fix one in production environment first)

**Comment by MaxGraey** [comment]
I investigate this unresolved issue.

It happening during parsing markdown. I think some issue like this happening we have got a lot of forceUpdate calls like in current [case](https://github.com/wasdk/WebAssemblyStudio/blob/97f21fc1dc3ddb7732b794ff8330005edf0b8046/src/components/App.tsx#L244). **forceUpdate** is antipattern and we should avoid this. Instead for main container and dialog use separate RenderDOMs or [Portals](https://reactjs.org/docs/portals.html)

**Comment by yurydelendik** [comment]
> I see using react-modal which already implement portal under the hood. So it very strange why we still needs in forceUpdate

Yes, we need to open an issue to remove/reduce forceUpdate usages
